packages/fluence-js/src/internal/FluencePeer.ts

```diff
@@ -633,19 +640,31 @@ export class FluencePeer {
                 }
 
                 // Do not continue if there was an error in particle interpretation
-                if (!isInterpretationSuccessful(item.result)) {
+                if (item.result instanceof Error) {
+                    log.error('Interpreter failed: ', jsonify(item.result.message));
+                    item.onStageChange({ stage: 'interpreterError', errorMessage: item.result.message });
+                    return;
+                }
+
+                const toLog = { ...item.result, data: dataToString(item.result.data) };
+                if (item.result.retCode !== 0) {
+                    log.error('Interpreter failed: ', jsonify(toLog));
                     item.onStageChange({ stage: 'interpreterError', errorMessage: item.result.errorMessage });
                     return;
+                } else {
+                    log.debug('Interpreter result: ', jsonify(toLog));
                 }
 
+                const newData = Buffer.from(item.result.data);
```

Comment:
also I think no need for newData variable. Just from reading this part of the code it seems like it's won't be used at all if item.result.nextPeerPks.length === 0, so I think it's better to just move it inside if statement directly